Source/Models/JSONDeserializer.swift

```diff
@@ -0,0 +1,30 @@
+import Argo
+import Result
+
+struct JSONDeserializer: Deserializer {
+  func deserialize<T : Request>(request: T)(data: NSData?) -> Result<T.Decoder.Representation, SwishError> {
```

Comment:
Actually, maybe this _shouldn't_ be curried at all, and we should just curry it at the call site using an internal implementation of `curry`.